client/my-sites/checkout/composite-checkout/wpcom/hooks/use-shopping-cart.ts

```diff
@@ -249,6 +252,24 @@ function shoppingCartHookReducer(
 	return state;
 }
 
+function removeItemFromLocalStorage( productSlug: string ) {
+	const cartItemsFromLocalStorage = JSON.parse(
+		window.localStorage.getItem( 'shoppingCart' ) || '[]'
+	);
+
+	if ( ! isEmpty( cartItemsFromLocalStorage ) ) {
+		const newCart = cartItemsFromLocalStorage.filter(
+			( product ) => product.product_slug !== productSlug
+		);
+
+		try {
+			window.localStorage.setItem( 'shoppingCart', JSON.stringify( newCart ) );
+		} catch ( e ) {
+			throw new Error( 'An unexpected error occured while saving your cart: ' + e );
```

Comment:
We should probably translate this error message if there's any chance it might be displayed to the user. If it is not displayed, we might want to do something to make sure it _does_ get displayed because if this save fails then I suspect the user is going to have problems going forward.